src/metabase/driver/postgres.clj

```diff
@@ -804,12 +804,13 @@
                                      :columns     (map keyword column-names)
                                      ::from-stdin "''"}
                                     :quoted true
-                                    :dialect (sql.qp/quote-style driver))
-           tsvs         (->> values
-                             (map row->tsv)
-                             (str/join "\n")
-                             (StringReader.))]
-       (.copyIn copy-manager ^String sql tsvs)))))
+                                    :dialect (sql.qp/quote-style driver))]
+       (doseq [slice-of-values (partition-all 100 values)]
```

Comment:
I suggest leaving a comment explaining why 100 was the magic number. And if you didn't try other numbers because it's good enough, leave a comment saying there's room for more optimization.